File: DONGUK/프로그래머스_WEEK 9/최소 직사각형.py
def solution(sizes):
    answer = 0

    orders = []
    for size in sizes:
        a = max(size[0], size[1])
        b = min(size[0], size[1])
        orders.append((a, b))
    
    garo, sero = 0, 0
    for a, b in orders:
        garo = max(garo, a)
        sero = max(sero ,b)
    
    answer = garo * sero
    return answer

File: JUNHEE/프로그래머스_week9/최소 직사각형.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int solution(vector<vector<int>> sizes) {
    int answer = 0;
    int maxlen = 0, minlen = 0;
    for(int i = 0; i < sizes.size(); i++){
        if(sizes[i][0] < sizes[i][1]){
            if(minlen < sizes[i][0]) minlen = sizes[i][0];
            if(maxlen < sizes[i][1]) maxlen = sizes[i][1];
        }
        else{
            if(minlen < sizes[i][1]) minlen = sizes[i][1];
            if(maxlen < sizes[i][0]) maxlen = sizes[i][0];
        }
    }
    answer = maxlen * minlen;
    return answer;
}
